SEBrowser/Controllers/OpenXDA/TrendChannelController.cs

```diff
@@ -0,0 +1,353 @@
+
+                string filters =
+                    $@"Channel.Trend = 1
+                    {(string.IsNullOrEmpty(phaseFilter) ? "" : $"AND ({phaseFilter})")}
+                    {(string.IsNullOrEmpty(channelGroupFilter) ? "" : $"AND ({channelGroupFilter})")}
+                    {(string.IsNullOrEmpty(meterFilter) ? "" : $"AND {meterFilter}")}
+                    {(string.IsNullOrEmpty(assetFilter) ? "" : $"AND {assetFilter}")}";
+
+                string query =
+                    $@"SELECT
+	                    Channel.ID,
+	                    Channel.Name,
+	                    Channel.Description,
+                        Asset.ID as AssetID,
+	                    Asset.AssetKey,
+	                    Asset.AssetName,
+                        Meter.ID as MeterID,
+	                    Meter.AssetKey AS MeterKey,
+	                    Meter.Name AS MeterName,
+                        Meter.ShortName AS MeterShortName,
+	                    Phase.Name AS Phase,
+	                    ChannelGroup.Name AS ChannelGroup,
+	                    ChannelGroupType.DisplayName AS ChannelGroupType,
+	                    ChannelGroupType.Unit
+                    FROM 
+	                    Channel LEFT JOIN
+	                    Phase ON Channel.PhaseID = Phase.ID LEFT JOIN
+	                    Asset ON Asset.ID = Channel.AssetID LEFT JOIN
+	                    Meter ON Meter.ID = Channel.MeterID LEFT JOIN
+	                    ChannelGroupType ON Channel.MeasurementCharacteristicID = ChannelGroupType.MeasurementCharacteristicID AND Channel.MeasurementTypeID = ChannelGroupType.MeasurementTypeID LEFT JOIN
+	                    ChannelGroup ON ChannelGroup.ID = ChannelGroupType.ChannelGroupID
+                    WHERE
+	                    {filters}";
+
+                DataTable table = connection.RetrieveData(query);
+
+                return table;
+            }
+        }
+
+        [Route("GetLineChartData"), HttpPost]
+        public IHttpActionResult GetLineChartData([FromBody] JObject postData)
+        {
+            Task<Stream> streamTask = XDAAPIHelper.Post("api/HIDS/QueryPoints", new StringContent(postData.ToString(), Encoding.UTF8, "application/json"));
+            IEnumerable<int> channelIds = postData["Channels"].ToObject<IEnumerable<int>>();
+            List<ChannelDataSet> channelList = channelIds.Select(id => new ChannelDataSet()
+            {
+                Points = new List<object>(),
+                ChannelID = id.ToString("x8")
+            }).ToList();
+            using (Stream stream = streamTask.Result)
+            using (TextReader reader = new StreamReader(stream))
+            {
+                while (true)
+                {
+                    string line = reader.ReadLine();
+
+                    if (line == null)
+                        break;
+
+                    if (line == string.Empty)
+                        continue;
+
```

Comment:
Do we really need to parse it? Wouldn't a `Jobject` be sufficient since we are sending it over to the client anyway?

Reply:
Good point, we don't need to I think.

Reply:
After Discussion with Stephen, the reason HIDDS uses a stream is because querying multiple years of data and puttingit into memory to build JSON may cause memory issues. 
Based on that it may be beneficial to move to streaming the data from here and using something like:
https://stackoverflow.com/a/18964123
to parse the stream client side.